packages/pixelboy/apps/journal/checkin/checkin.coffee

```diff
@@ -18,12 +18,14 @@ class PAA.PixelBoy.Apps.Journal.CheckIn extends AM.Component
   events: ->
     super.concat
       'submit .check-in-form': @onSubmitCheckInForm
+      'input .external-url': @getExternalImgUrl
+      'change .image-file': @generateLocalImagePreview
```

Comment:
`@onChangeImageFile` and from there you can call `@generateLocalImagePreview()`. The idea is that we always assume the handler is always specific to handle that event. The thinking would go like this: what if on change image file you also then had to add some other side effect that has nothing to do with generating a local image preview. Then you'd probably go and refactor and rename. So here we're unambiguously taking care of this preventively.